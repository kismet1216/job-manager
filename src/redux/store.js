import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga'
import rootReducer from './reducers';
import mySaga from './saga';

// create the saga middleware
const saga = createSagaMiddleware();
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
  rootReducer,
  undefined,
  composeEnhancers(
    applyMiddleware(
      saga
    )
  )
);

// then run the saga
saga.run(mySaga);

export default store;
