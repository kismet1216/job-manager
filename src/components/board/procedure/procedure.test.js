import React from 'react';
import Procedure from './procedure';
import renderer from 'react-test-renderer';
import Popover from '../../../shared/components/popover/popover';
import store from '../../../redux/store';
import { Provider } from 'react-redux';

describe('Procedure', () => {
  const info = {id: 1, title: '1'};

  it('has a valid menu snapshot', () => {
    const testRenderer = renderer.create(<Provider store={store}><Procedure info={info} /></Provider>);
    const testInstance = testRenderer.root;
    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();

    // trigger click
    testInstance.findByType(Popover).findByProps({className: 'trigger'}).props.onClick();
    tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
