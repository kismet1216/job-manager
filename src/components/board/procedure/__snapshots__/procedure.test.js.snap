// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Procedure has a valid menu snapshot 1`] = `
<div
  className="procedure border rounded text-center"
>
  <div
    className="d-flex justify-content-between align-items-center"
  >
    <button
      className="btn btn-link"
      onClick={[Function]}
    >
      <i
        className="fa fa-plus"
      />
    </button>
    <div
      onClick={[Function]}
    >
      1
    </div>
    <div
      className="popover-component"
    >
      <div
        className="trigger"
        onClick={[Function]}
      >
        <button
          className="btn btn-link"
        >
          <i
            className="fa fa-bars"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    className="cards-container"
    onDragOver={[Function]}
    onDrop={[Function]}
  />
</div>
`;

exports[`Procedure has a valid menu snapshot 2`] = `
<div
  className="procedure border rounded text-center"
>
  <div
    className="d-flex justify-content-between align-items-center"
  >
    <button
      className="btn btn-link"
      onClick={[Function]}
    >
      <i
        className="fa fa-plus"
      />
    </button>
    <div
      onClick={[Function]}
    >
      1
    </div>
    <div
      className="popover-component"
    >
      <div
        className="trigger"
        onClick={[Function]}
      >
        <button
          className="btn btn-link"
        >
          <i
            className="fa fa-bars"
          />
        </button>
      </div>
      <div
        className="popover-content border rounded"
        style={
          Object {
            "width": "200px",
          }
        }
      >
        <div
          className="list-group list-group-flush"
        >
          <button
            className="list-group-item list-group-item-action"
          >
            左移
          </button>
          <button
            className="list-group-item list-group-item-action"
          >
            右移
          </button>
          <button
            className="list-group-item list-group-item-action"
          >
            时间先后排序
          </button>
          <button
            className="list-group-item list-group-item-action"
          >
            时间先后逆序
          </button>
          <button
            className="list-group-item list-group-item-action"
          >
            优先级排序
          </button>
          <button
            className="list-group-item list-group-item-action list-group-item-danger"
            onClick={[Function]}
          >
            清空列表
          </button>
          <button
            className="list-group-item list-group-item-action list-group-item-danger"
            onClick={[Function]}
          >
            删除列表
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="cards-container"
    onDragOver={[Function]}
    onDrop={[Function]}
  />
</div>
`;
